pkg/common/cns-lib/vsphere/hostsystem.go

```diff
@@ -219,3 +219,28 @@ func GetSharedDatastoresForHosts(ctx context.Context, hosts []*HostSystem) ([]*D
 	}
 	return sharedDatastores, nil
 }
+
+// GetAllAccessibleDatastoresForHosts returns datastores accessible to hosts mentioned in the input parameter.
+func GetAllAccessibleDatastoresForHosts(ctx context.Context, hosts []*HostSystem) ([]*DatastoreInfo, error) {
+	log := logger.GetLogger(ctx)
+	var allAccessibleDatastores []*DatastoreInfo
+
+	for _, host := range hosts {
+		accessibleDatastores, err := host.GetAllAccessibleDatastores(ctx)
+		if err != nil {
+			log.Warnf("failed to fetch datastores from host %+v. Error: %+v",
+				host, err)
+			continue
+		}
+		if len(accessibleDatastores) == 0 {
+			log.Infof("No datastores found for host %+v", host)
+		} else {
+			allAccessibleDatastores = append(allAccessibleDatastores, accessibleDatastores...)
```

Comment:
Can we filter out duplicate datastores. we may have same datastore mounted across all hosts. function should not return datastore list with duplicate values.